src/raw/aws/fargate_metadata.go

```diff
@@ -172,3 +172,14 @@ func MetadataV4BaseURL() (*url.URL, error) {
 	}
 	return parsedURL, nil
 }
+
+// GetMetadataBaseURL returns metadataV4BaseURL if available, otherwise try with metadataV3BaseURL
+func GetMetadataBaseURL() (*url.URL, error) {
+	metadataBaseURL, err := metadataV4BaseURL()
+	if err == nil {
+		return metadataBaseURL, nil
+	}
+	log.Debug("The Metadata endpoint V4 is not available, falling back to V3: %s", err.Error())
+	// If we do not find V4 we fall back to V3
+	return metadataV3BaseURL()
```

Comment:
both functions does exactly the same , perhaps this could be just one func with the env var as parameter , or even do the logic here IMO it doesn't have the complexity worth the layering

Reply:
done!